### PR TITLE
Extend GeometryUtils, ArrayUtils

### PR DESCRIPTION
**Pull request by Gustl22**
GeometryUtils:
- isClockwise check for polygons
- area with positive value

ArrayUtils:
- reverse array for several data types

### CODE REVIEW

**Comment by Gustl22** [comment]
Needed for optimizations in #663, too

File: vtm/src/org/oscim/tiling/source/LwHttp.java [hunk 1]
```diff
@@ -457,7 +457,7 @@ private static int writeInt(int val, int pos, byte[] buf) {
         for (int n = val; n > 0; n = n / 10, i++)
             buf[pos + i] = (byte) ('0' + n % 10);
 
-        ArrayUtils.reverse(buf, pos, pos + i);
+        ArrayUtils.reverse(buf, pos, pos + i, 1);
```
Comment (devemux86) [inline vtm/src/org/oscim/tiling/source/LwHttp.java]: Why is that change needed?

**Comment by devemux86** [comment]
Thanks, merged via 1f0b5509ed4232570611cba34c9d7aa6823a0df6.
